### ISSUE TITLE
Usage of the `sct_analyze_lesion -f` option is unclear (percentage distributions)

### ISSUE THREAD

**Issue by jqmcginnis** [spinalcordtoolbox/spinalcordtoolbox#4394]
Hi all, 

I am currently working on a large-scale MS lesion analysis project using whole spine scans for which I (1) register all images to the PAM50 (2) warp PAM50 atlases to the patient space (3) run sct_analyze_lesions using the lesions in patient space and the warped template. After taking a thorough look at the analyze_lesion_results, I suspect a bug in the analysis pipeline.

### Steps to Reproduce

1. Using the newest SCT v6.2 release with (updated) PAM50 Template, please run the following command:

`sct_analyze_lesion -m lesion_mask.nii.gz -f /path/to/warped/PAM50 -i /path/to/image`

2. Having a look at the results, specifically the xls file generated by the command, we have noticed some unexpected things:

The xls file typically consists of multiple sheets, i.e. measures, lesion1#_distribution, lesion2#_distribution, ..., lesionN#_distribution, and finally ROI_occupied_by_lesion.

- Given the lesion volume statistics in the "measure" sheets, we notice that they do not correspond to the total lesion volumes in the corresponding lesion#N_distribution sheet. This may be the case as the individual atlas labels might be overlapping, i.e. the intersection is not exclusive to one region, and thus may result in higher intersection summation results? However, we also notice that the total volume sum is wrong (even within a sheet), specifically it is always "100".

- Given the miscalculated lesion volume sheets, the "ROI_occupied_by_lesion" seems off as well (especially if compared to the measures stats.) We are unsure if this is related to the preceding analysis, and thus an effect of a bug, or is a separate issue.

2. To reproduce results of all pipeline stages I have shared slected images (and the entire pipeline script) via a link privately shared via slack. Otherwise, have a look at the xls file I have attached here :+1: 
[tum_acq-ax_T2w_lesion_analysis.xls](https://github.com/spinalcordtoolbox/spinalcordtoolbox/files/14526657/tum_acq-ax_T2w_lesion_analysis.xls)

Please let me know, how I can assist :slightly_smiling_face: 

-Julian

**Comment by joshuacwnewton** [comment]
Thank you so much for reporting this issue, and my apologies for the delay in response! I plan to take a look at this today. :slightly_smiling_face:

**Comment by joshuacwnewton** [comment]
Given the data you shared privately, I chose `sub-m058xxx_ses-2021xxxx` to start. The command I ran was:

```bash
sct_analyze_lesion -m "dataset/derivatives/labels/sub-m058xxx/ses-2021xxxx/anat/sub-m058xxx_ses-2021xxxx_acq-ax_T2w_lesion-manual.nii.gz"
                   -i "dataset/sub-m058xxx/ses-2021xxxx/anat/sub-m058xxx_ses-2021xxxx_acq-ax_T2w.nii.gz"
                   -f "pipeline_results/data_processed/derivatives/sct/sub-m058xxx/ses-2021xxxx/anat/"
```

> [!NOTE]
> As a side observation, I noticed that adding the `-f` option makes `sct_analyze_lesion` _very_ slow on my personal laptop, even for a single subject. There is no feedback in the output while the template-related processing is running, too. This isn't a great user experience, and I would love to look further into improving this after I debug the numerical issues.

----

Anyway, as far as the results go...

> Given the lesion volume statistics in the "measure" sheets, we notice that they do not correspond to the total lesion volumes in the corresponding lesion#N_distribution sheet. This may be the case as the individual atlas labels might be overlapping, i.e. the intersection is not exclusive to one region, and thus may result in higher intersection summation results? However, we also notice that the total volume sum is wrong (even within a sheet), specifically it is always "100".

When looking at the generated output .xls file, I was able to confirm the behavior you describe. So, I then hooked up a debugger and traced the script, and found this section:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/44587d4cb293e04e635e17fe56137a0c4aa28f8f/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L377-L382

It looks like "summing to 100" is intentional -- these values are percentages!

> Given the miscalculated lesion volume sheets, the "ROI_occupied_by_lesion" seems off as well (especially if compared to the measures stats.) We are unsure if this is related to the preceding analysis, and thus an effect of a bug, or is a separate issue.

The ROI values _also_ seem to be percentages! 

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/44587d4cb293e04e635e17fe56137a0c4aa28f8f/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L328-L334

----

All together, it seems as though the intended usage of this `-f` option is not entirely clear?

(In fact, to be candid, I have not actually seen the `-f` option be used before since I joined SCT in 2020. The option is not mentioned in the tutorials, we have no tests for it, and it is also not included in `batch_processing.sh`. So, there is definitely room to improve this option to make it more useful for our users!) 

When I look at the history of the `sct_analyze_lesion` script, it seems as though its implementation is mostly unchanged from how it was when first implemented 7 years ago: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/1439. So, perhaps some insight into its intended usage can be gleamed from the original design discussions: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/1351, specifically https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/1351#issuecomment-318519269? We can see some sample "interpretations" of the sheet tabs in that comment.

With that in mind, if there is anything we can do to improve the documentation for this option, and/or make it more useful to your group, please let us know!

**Comment by joshuacwnewton** [comment]
((Note to self: @jqmcginnis is on vacation right now, so it's OK and expected for this issue to be quiet in the short-term.))

**Comment by jqmcginnis** [comment]
@joshuacwnewton I have had a thorough look at the discuss in #1351 and that helped a lot to clarify my questions. Thank you very much for looking into the details, that was very helpful!

Before we proceed to discuss the documentation part, I was wondering if you can help me understand the "total" row in l.22 of the file I have provided? :slightly_smiling_face: 

If we take the file I have uploaded in the issue and examine sheet lesion#1_distribution e.g. B20 (i.e. approx 11% of Tract 17 is occupied by lesion in level 19) and approx. 94.4% of the same tract (i.e. B21). Can you explain how these value can be summed up in line B22 - is this meaningful? Or what's the intuition behind it? I am afraid I do not understand that part or overlooked the explanation. 

Thank you once again!

**Comment by joshuacwnewton** [comment]
> If we take the file I have uploaded in the issue and examine sheet lesion#1_distribution e.g. B20 (i.e. approx 11% of Tract 17 is occupied by lesion in level 19) and approx. 94.4% of the same tract (i.e. B21). Can you explain how these value can be summed up in line B22 - is this meaningful? Or what's the intuition behind it? I am afraid I do not understand that part or overlooked the explanation.

I would be happy to explain! (tl;dr: `0.11` refers to `0.11%`, rather than `11%`

Just for reference, here is the sheet as a whole, but with additional border/color formatting:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/82adedf3-09f0-44f1-95a7-f5cecd20bd3c)

I believe the interpretation of these values is as follows:

- `B20`:`0.11` --> 0.11% of the lesion is in the "PAM50_17" tract, in level 19.
- `B` column --> `B2:B21` sums to the "total" value in `B22` --> 1.06% of the lesion is in the "PAM50_17" tract.
- `20` row -> `B20:AK20` sums to the "total" value in `AL20` --> 30.94% of the lesion is in level 19.

In other words, the values in the range `B2:AK21` can be summed either **horizontally** (totals: column `AL`) or **vertically** (totals: row `22`). And, if you sum both horizontally _and_ vertically, you will sum to 100 (bottom-right cell `AL22`).

**Comment by jqmcginnis** [comment]
@joshuacwnewton Sorry for the late reply, and thank you very much for the repeated clarification and patience, I cannot believe it took me that long to understand what is actually going on in the sheet :smile: but I believe everything is clear to me now.

Should we add something to the documentation? Perhaps in the total columns and rows, to explain what's going on?

**Comment by jcohenadad** [comment]
> Perhaps in the total columns and rows, to explain what's going on?

to discuss during next SCT meeting

**Comment by joshuacwnewton** [comment]
I can draft a PR to get the ball rolling, and then we can discuss the changes in the meeting if need be. :)
